internal/test/e2e/e2e_test.go

```diff
@@ -162,88 +186,112 @@ func TestE2E_Kind(t *testing.T) {
 		return nil
 	}, time.Minute, time.Second*5)
 	require.NoError(t, err)
-
-	t.Log("Running tests")
-	runAllTests(t, kubectl)
 }
 
-func TestE2E_CurrentContext(t *testing.T) {
-	if !*fUseCurrentContext {
-		t.Skip()
-	}
-
-	home, err := os.UserHomeDir()
-	require.NoError(t, err)
-	configPath := filepath.Join(home, ".kube", "config")
-	if path, found := os.LookupEnv("KUBECONFIG"); found {
-		configPath = path
-	}
-
-	kubectl := &kubectlContext{
-		t:          t,
-		configPath: configPath,
-	}
-
-	runAllTests(t, kubectl)
-}
-
-// Starts a Kind cluster on the local machine, exposing port 2379 accepting ETCD connections.
-func setupLocalCluster(t *testing.T) (*cluster.Context, func()) {
-	t.Log("Starting Kind cluster")
-	kind := cluster.NewContext("etcd-e2e")
-
-	err := kind.Create(create.WithV1Alpha3(&kindv1alpha3.Cluster{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "Cluster",
-			APIVersion: "kind.sigs.k8s.io/v1alpha3",
-		},
-		Nodes: []kindv1alpha3.Node{
-			{
-				Role: "control-plane",
-				ExtraPortMappings: []cri.PortMapping{
-					{
-						ContainerPort: 32379,
-						HostPort:      2379,
-					},
-				},
-			},
-		},
-	}))
-	require.NoError(t, err)
-
-	tearDown := func() {
-		if !*fCleanup {
-			return
+func TestE2E(t *testing.T) {
+	var (
+		kubectl *kubectlContext
+	)
+	switch {
+	case *fUseKind:
+		var (
+			kind     *cluster.Context
+			stopKind func() error
+			imageTar string
+			wg       sync.WaitGroup
+		)
+		defer func() {
+			if stopKind != nil {
+				err := stopKind()
+				require.NoError(t, err)
+			}
+		}()
+		ctx, cancel := context.WithCancel(context.Background())
```

Comment:
i put a 5 minute timeout on this initially, any reason that's gone?

Reply:
I may have misunderstood.
But in master, the 5 minute context only seems to be used for etcdclient operations.
It is started *after* the Kind cluster has been started.
This context has no timeout and  is used to help quit the the other setup related go routine if either of them fails.
So if the `buildOperator` function fails the kind cluster will continue to completion but the test will then exit.
kind.Create doesn't accept a context; the api has no cancellation  ASFAICS.
If the Kind cluster startup fails (which it does if there is already a Kind cluster running) then the `buildOperator` go routine will also be cancelled at the next `exec.CommandContext` call and the test will exit early.

Reply:
So I'll leave it as-is for now.